lib/Wikia/src/Util/PerformanceProfilers/UsernameUseProfiler.php

```diff
@@ -42,28 +52,44 @@ class UsernameUseProfiler {
 	 * UsernameUseProfiler constructor.
 	 * @param $class string
 	 * @param $method string
-	 * @param $sampling float
+	 * @param $traceId string
+	 * @param $shouldSample boolean
 	 */
-	public function __construct( $class, $method, $sampling = 0.01 ) {
+	private function __construct( $class, $method, $traceId, $shouldSample ) {
 		$this->class = $class;
 		$this->method = $method;
-		$this->sampling = $sampling;
+		$this->traceId = $traceId;
+		$this->shouldSample = $shouldSample;
 		$this->start();
 	}
 
-	public function start() {
-		$this->startTime = $this->startProfile();
+	private function start() {
+		if( $this->shouldSample ) {
+			$this->startTime = $this->startProfile();
+		}
 	}
 
 	public function end( $context = [] ) {
-		$context[ 'class' ] = $this->class;
-		$context[ 'method' ] = $this->method;
-
-		$this->endProfile(
-			static::EVENT_USERNAME_FROM_ID,
-			$this->startTime,
-			$context,
-			$this->sampling
-		);
+		if( $this->shouldSample ) {
+			$context['class'] = $this->class;
+			$context['method'] = $this->method;
+			$context['trace_id'] = $this->traceId;
+
+			$this->endProfile(
+				static::EVENT_USERNAME_FROM_ID,
+				$this->startTime,
+				$context,
+				$this->sampling
+			);
+		}
+	}
+
+	public static function create( $class, $method ) {
+		if ( static::$shouldSampleRequest === null ) {
```

Comment:
I would add a line of comment what was our intention - enable profiling on a per request basis. Otherwise I can imagine this construct can be difficult to understand without context.